homeassistant/components/device_tracker/tilgin.py

```diff
@@ -0,0 +1,142 @@
+"""
+Support for Tilgin routers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.tilgen/
+"""
+
+import re
+import hmac
+import logging
+import hashlib
+import requests
+
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import (
+    DOMAIN, PLATFORM_SCHEMA, DeviceScanner)
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_USERNAME)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['beautifulsoup4==4.6.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+HTTP_HEADER_NO_CACHE = 'no-cache'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string
+})
+
+
+def get_scanner(hass, config):
+    """Validate the configuration and return a Tilgin Device Scanner."""
+    models = [TilginHG238xDeviceScanner]
+    for model_family in models:
+        scanner = model_family(config[DOMAIN])
+        if scanner.success_init:
+            return scanner
+
+    return None
+
+
+class TilginHG238xDeviceScanner(DeviceScanner):
+    """Queries the router for connected devices."""
+
+    def __init__(self, config):
+        """Initialize the scanner."""
+        self.url = 'http://{}'.format(config[CONF_HOST])
+        self.username = config[CONF_USERNAME]
+        self.password = config[CONF_PASSWORD]
+        self.session = requests.Session()
+
+        self.last_results = {}
+        self.success_init = False
+
+        _LOGGER.debug("Initialising Tilgin HG238x Device")
+        self.success_init = self._check_auth()
+
+    def scan_devices(self):
+        """Scan for new devices and return a list with found device IDs."""
+        self._update_info()
+        return self.last_results.keys()
+
+    def get_device_name(self, device):
+        """Get the name of the device."""
+        return self.last_results.get(device)
+
+    def _check_auth(self):
+        """Check the credentials work."""
+        self._authenticate()
+        res = self.session.get(self.url)
```

Comment:
The requests and data parsing should be moved to a standalone library published on pypi.